pinot-spi/src/main/java/org/apache/pinot/spi/recordenricher/RecordEnricher.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.recordenricher;
+
+import java.util.List;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+public interface RecordEnricher {
+  List<String> getInputColumns();
+  void enrich(GenericRow record);
```

Comment:
Can you add some javadoc?

I would like to know at least what happens if the enriched column is already in the GenericRow.

Reply:
Added java docs